src/store/actions.js

```diff
@@ -1,46 +1,88 @@
-import Xo from 'xo-lib'
-import { createBackoff } from 'jsonrpc-websocket-client'
-
-// ===================================================================
-
-const createAction = (() => {
-  const { defineProperty } = Object
-  const identity = payload => payload
-
-  return (type, payloadCreator = identity) => defineProperty(
-    (...args) => ({
-      type,
-      payload: payloadCreator(...args)
-    }),
-    'toString',
-    { value: () => type }
-  )
-})()
-
-// ===================================================================
-
-const xo = new Xo({
-  url: 'localhost:9000'
-})
-{
-  const connect = () => {
-    xo.open(createBackoff()).catch(error => {
-      console.error('failed to connect to xo-server', error)
-    })
+/*import Xo from 'xo-lib'
+import { createBackoff } from 'jsonrpc-websocket-client'*/
+
+/* action type */
+export const VM_CREATE = 'VM_CREATE'    // create a VM in store
+export const VM_EDIT = 'VM_EDIT'        // edit a vm in store
+export const VM_SAVE = 'VM_SAVE'        // want to save a vm from store to server
+export const VM_SAVED = 'VM_SAVED'      // VM is saved on server
+
+export const SIGN_IN = 'SIGN_IN'     // ask to signin
+export const SIGNED_IN = 'SIGNED_IN'    // is signed in
+export const SIGN_OUT = 'SIGN_OUT'   // want to sign out
+export const SIGNED_OUT = 'SIGNED_OUT'  // signed out
+export const SESSION_PATCH = 'SESSION_PATCH'  // signed out
+
+/* action creator
+ * they HAVE TO return an action with the mandatory field type, and an optiona payload
+ * they MAY dispatch ( emit ) other actions, async or not
+ * action will be used by reducers
+ */
+export function patchSession (patch) {
+  return {
+    type: SESSION_PATCH,
+    payload: patch
   }
-  xo.on('scheduledAttempt', ({ delay }) => {
-    console.log('next attempt in %s ms', delay)
-  })
+}
+export function signIn () {
+  // using redux thunk https://github.com/gaearon/redux-thunk
+  // instead of returning one promise, it can dispatch multiple events
+  // that way it become trivial to inform user of progress
+  return dispatch => {
+    setTimeout(() => {
+      // Yay! Can invoke sync or async actions with `dispatch`
+      dispatch(signedIn({userId: Math.floor(Math.random() * 1000)}))
+    }, 10)
 
-  xo.on('closed', connect)
-  connect()
+    dispatch({type: 'SIGN_IN'}) // immediatly inform the sore that we'll try to signin
+  }
```

Comment:
For async actions, I would prefer using promises (with `redux-promise`) which feels more standard and should alleviate the needs to create new actions manually (for success/error).